no fseeko() and ftello() in ppc-morphos. add a makefile for ppc-morphos.

diff -u flac-1.3.x-dec/include/share/compat.h~ flac-1.3.x-dec/include/share/compat.h
--- flac-1.3.x-dec/include/share/compat.h~
+++ flac-1.3.x-dec/include/share/compat.h
@@ -66,6 +66,10 @@
 #define fseeko fseek
 #define ftello ftell
 #define FLAC__off_t off_t
+#elif defined(__MORPHOS__)
+#define fseeko fseek
+#define ftello ftell
+#define FLAC__off_t off_t
 #else
 #define FLAC__off_t off_t
 #endif
diff -u /dev/null flac-1.3.x-dec/src/libFLAC/Makefile.morphos
--- /dev/null
+++ flac-1.3.x-dec/src/libFLAC/Makefile.morphos
@@ -0,0 +1,42 @@
+ifeq ($(CROSS),)
+CC=gcc
+AS=as
+AR=ar
+RANLIB=ranlib
+else
+CC=$(CROSS)-gcc
+AS=$(CROSS)-as
+AR=$(CROSS)-ar
+RANLIB=$(CROSS)-ranlib
+endif
+
+INCLUDES=-I. -I./include -I../../include
+
+CFLAGS =-noixemul -O2 -Wall -W
+#CFLAGS+=-maltivec -mabi=altivec
+
+CPPFLAGS=-DHAVE_STDINT_H -DHAVE_INTTYPES_H -DHAVE_SYS_PARAM_H
+CPPFLAGS+= -DWORDS_BIGENDIAN=1 -DFLAC__CPU_PPC=1 -DSIZEOF_VOIDP=4
+#CPPFLAGS+= -DFLAC__ALIGN_MALLOC_DATA
+CPPFLAGS+= -DNDEBUG
+CPPFLAGS+= -DENABLE_64_BIT_WORDS=0
+CPPFLAGS+= -DFLAC__HAS_OGG=1
+CPPFLAGS+= -DPACKAGE_VERSION=\"1.3.0\"
+
+COMPILE=$(CC) $(CFLAGS) $(CPPFLAGS) $(INCLUDES)
+
+OBJ=bitmath.o bitreader.o cpu.o crc.o fixed.o format.o lpc.o md5.o memory.o stream_decoder.o ogg_decoder_aspect.o ogg_mapping.o
+
+all: libFLAC.a
+
+libFLAC.a: $(OBJ)
+	$(AR) cru $@ $(OBJ)
+	$(RANLIB) $@
+
+%.o: %.c
+	$(COMPILE) -c -o $@ $<
+
+clean:
+	rm -rf *.o
+distclean: clean
+	rm -rf *.a
